tests/integration/long/test_failure_types.py

```diff
@@ -0,0 +1,295 @@
+                "Native protocol 4,0+ is required for custom payloads, currently using %r"
+                % (PROTOCOL_VERSION,))
+
+        self.cluster = Cluster(protocol_version=PROTOCOL_VERSION)
+        self.session = self.cluster.connect()
+        self.nodes_currently_failing = []
+        self.node1, self.node2, self.node3 = get_cluster().nodes.values()
+
+    def tearDown(self):
+
+        self.cluster.shutdown()
+        failing_nodes = []
+
+        # Restart the nodes to fully functional again
+        self.setFailingNodes(failing_nodes, "testksfail")
+
+    def setFailingNodes(self, failing_nodes, keyspace):
+        """
+        This method will take in a set of failing nodes, and toggle all of the nodes provided to fail writes. It will
+        @param failing_nodes A definitive list of nodes that should fail writes
+        @param keyspace The keyspace to enable failures on
+
+        """
+
+        # Ensure all of the nodes on the list have failures enabled
+        for node in failing_nodes:
+            if node not in self.nodes_currently_failing:
+                print "Stopping node "+node.name
+                node.stop(wait_other_notice=True, gently=False)
+                print "starting crippled node "+node.name
+                node.start(jvm_args=[" -Dcassandra.test.fail_writes_ks=" + keyspace], wait_for_binary_proto=True,
+                           wait_other_notice=True)
+                self.nodes_currently_failing.append(node)
+
+        # Ensure all nodes not on the list, but that are currently set to failing are enabled
+        for node in self.nodes_currently_failing:
+            if node not in failing_nodes:
+                print "stopping node "+node.name
+                node.stop(wait_other_notice=True, gently=False)
+                print "starting node "+node.name
+                node.start(wait_for_binary_proto=True, wait_other_notice=True)
+                self.nodes_currently_failing.remove(node)
+
+    def _perform_cql_statement(self, text, consistency_level, expected_exception):
+        """
+        Simple helper method to preform cql statements and check for expected exception
+        @param text CQl statement to execute
+        @param consistency_level Consistency level at which it is to be executed
+        @param expected_exception Exception expected to be throw or none
+        """
+        statement = self.session.prepare(text)
+        statement.consistency_level = consistency_level
+
+        if expected_exception is None:
+            self.session.execute(statement)
+        else:
+            with self.assertRaises(expected_exception) as cm:
+                self.session.execute(statement)
+
+    def test_write_failures_from_coordinator(self):
+        """
+        Test to validate that write failures from the coordinator are surfaced appropriately.
+
+        test_write_failures_from_coordinator Enable write failures on the various nodes using a custom jvm flag,
```

Comment:
I don't see anything guaranteeing that the write failure is generated on the coordinator. I do see that it's generated on node1, which might be the coordinator, depending on the query plan generated.
This is fine, but just making sure the test intent matches what is happening.